src/test/cache/cache_test.cpp

```diff
@@ -246,7 +246,8 @@ TEST(CachePolicyTest, Iterators) {
   auto element_count = size_t{0};
   auto value_sum = size_t{0};
 
-  for (const auto& [key, value] : cache) {
+  for (auto it = cache.unsafe_begin(); it != cache.unsafe_end(); ++it) {
```

Comment:
So the rename of the method leads to not being able to use the range-loops anymore? It might be just me, but I would have thought that it's clear that iterating a cache should be used with great caution.

Reply:
> So the rename of the method leads to not being able to use the range-loops anymore?

Yes.

> I would have thought that it's clear that iterating a cache should be used with great caution.

I believe that someone looking at the class and seeing a mutex being used for get/set is right to believe that the class transparently synchronizes accesses. Mixing thread-safe and -unsafe methods in one class is almost a guarantee for disaster. In my opinion, using range loops is by far the lesser evil.

Reply:
Sounds reasonable to me.